### ISSUE TITLE
Paramétrage de l'activation des couches d'un thème et affichage panneau au clic à droite ou en bas ?

### ISSUE THREAD

**Issue by frgis** [mviewer/mviewer#774]
Bonjour,

nous sommes en version 3.9.1 de Mviewer.

j'ai deux questions si quelqu'un peut m'aider :

1- Dans la doc de mviewer https://mviewerdoc.readthedocs.io/fr/latest/doc_tech/config_app.html il y a le paramètre togglealllayersfromtheme de documenté.
Sur une carte test j'ai essayé de l'ajouter à true mais je ne vois pas de différence par rapport à false ?
On est d'accord que normalement il devrait y avoir une case à cocher sur le thème permettant d'activer/désactiver ce qu'il y a à l'intérieur ?

Quelqu'un a un exemple consultable qui fonctionne, avec le xml ?

2- Je ne comprends pas comment on configure une carte pour que le panneau de consult s'affiche en bas et pas à droite ?

Merci !

**Comment by Agath21** [comment]
Bonjour, 

### Question 1 
En activant le paramètre togglealllayersfromtheme="true", vous devriez vois apparaître un badge à côté du nom de la thématique ouverte ressemblant à cela : 
![image](https://github.com/mviewer/mviewer/assets/22764056/22ba9bbb-565f-4e27-b4a9-3d8223b7ac93)

Si on clique sur le badge, il est possible d'afficher/masquer toutes les couches de la thématique à la fois. Vous pouvez tester le fonctionnement dans cette démo : https://kartenn.region-bretagne.fr/kartoviz/?config=demo/theme-designfr/theme-designfr.xml# 

### Question 2 
Pour utiliser le bottom-panel lors de l'interrogation d'une couche, il faut configurer le `layer` avec ce paramètre `infopanel="bottom-panel"`

Exemple : 
```
<layer id="rp_struct_pop_geom" name="Chiffre + 2 Diagrammes - Template 7.3" visible="false" tiled="false" exclusive="true" searchable="false" queryable="true" infopanel="bottom-panel" infoformat="application/vnd.ogc.gml" featurecount="1" legendurl="demo/mviewer_templates/img/legendurl_templates.png" filter="level='EPCI'" url="https://ows.region-bretagne.fr/geoserver/rb/wms" attribution="Sources: INSEE (RP) - OpenStreetMap | Traitements: Région Bretagne - Service connaissance, observation, planification et prospective" metadata="https://kartenn.region-bretagne.fr/geonetwork/srv/fre/catalog.search#/metadata/26324529-e0b7-450c-9506-2dcdca608f5f" metadata-csw="https://kartenn.region-bretagne.fr/geonetwork/srv/eng/csw?SERVICE=CSW&VERSION=2.0.2&REQUEST=GetRecordById&elementSetName=full&ID=26324529-e0b7-450c-9506-2dcdca608f5f">
<template url="demo/mviewer_templates/templates/template.bottomGraphChiffres_2.mst"/>
</layer>
```

Vous pouvez vous aider de cette démo : https://kartenn.region-bretagne.fr/kartoviz/?config=demo/mviewer_templates/config.xml 

> Effectivement, après vérification, le paramètre info-panel n'est pas documenté , oublie de notre part ;)

**Comment by frgis** [comment]
Bonjour,

merci pour votre réactivité !

Effectivement, j'ai même pas essayé de cliquer sur le chiffre à droite ... je m'attendais à un bouton radio à gauche du thème comme pour les couches à l'intérieur du thème.
On est obligés de déplier le thème pour cocher ou décocher tout ce qu'il y a dedans.
J'ai un peu du mal à comprendre pourquoi ce choix ?

Pour le volet du bas ok ça fonctionne, effectivement j'avais pas trouvé de paramètre dans la doc mais il me semblait être tombée sur une carte sur laquelle ça marchait, merci !

**Comment by Gaetanbrl** [comment]
> J'ai un peu du mal à comprendre pourquoi ce choix ?

Je ne saurai expliquer ce choix (pour le moment il n'avait pas été questionné).
Cela dit,  je trouve le fonctionnement actuel assez esthétique avec des informations utiles que pour le thème sélectionné. Cela évite d'avoir des éléments cliquables partout.

Cela reste aussi cohérent avec le fonctionnement de sélection d'un thème (au clic) pour voir le contenu et gérer son affichage (visibilité à la couche ou pour toutes).

**Comment by frgis** [comment]
Bonjour,
Personnellement je trouve pas ça intuitif, en fait avant que je vous ai posé la question cette semaine je l'avais activé sans m'en rendre compte, j'ai vu 0/1 j'ai même pas réfléchis à cliquer je me suis dis, il doit manquer quelque chose.

En effet, si on propose à l'utilisateur d'activer une couche avec un bouton radio à gauche il m semble qu'il faut que ce soit pareil pour le thème. Sinon on perd la moitié des utilisateurs.
J'avoue personnellement ne pas comprendre l'intérêt de savoir combien de couches il y a dans un thème je ne vois pas quel utilisateur a besoin de le savoir. Dans mon cas ce qui m'intéressais c'était de tester et de me dire voyons voir si cette fonctionnalité répond au cas ou une personne veut savoir tous les sentiers de randonnée pédestre qu'il y a à proximité alors il coche le thème plutôt que chaque circuit un par un, pas plus que ça.

D'autre part, je crois qu'il y a un dysfonctionnement, j'ai essayé : j'ai un thème appelé "Randonnée pédestre" avec des layer constituant les différents sentiers de randonnées proposés dans le thème (je crois qu'il y en a 5 de mémoire).
Quand je démarre la carte le thème écrit 0/1 au lieu de 0/5, c'est quand je clique pour déplier (je ne sais plus si c'est quand j'en active au moins 1 à l'intérieur du thème) qu'il écrit 0/5, il compte pas à l'initialisation de la carte. Vous pensez à un bug ou c'est normal ?

Mis à part ça c'est des choix de développement, ça ne correspond pas pour notre cas à ce que l'on a besoin mais cela doit se justifier pour d'autres cartes.
Merci pour votre aide, vous me direz si c'est utile que je déclare un dysfonctionnement à l'initialisation de la carte.

Bon weekend

**Comment by Gaetanbrl** [comment]
@lecault je te laisse classer cette issue selon bug ou question ou enhancement

**Comment by frgis** [comment]
Bonjour,

merci pour votre retour.
La carte sur laquelle j'ai découvert cette anomalie n'a pas de groupe tel que vous l'entendez mais seulement plusieurs "theme" et "layer" dedans.

![Capture d’écran 2023-08-21 082129](https://github.com/mviewer/mviewer/assets/56657014/a1d33206-3087-43f9-9702-94ed84f4e914)

A vérifier de plus près, j'ai l'impression que le problème arrive quand on a un thème qui content plusieurs layer mais aucun activé, ci-dessous une illustration :

![Capture d’écran 2023-08-21 082319](https://github.com/mviewer/mviewer/assets/56657014/e218d5ab-6374-429d-ab26-00994287be9e)


![Capture d’écran 2023-08-21 082407](https://github.com/mviewer/mviewer/assets/56657014/97c50103-9f27-42e0-9900-9fa379254ccc)

Vous reproduisez dans ce cas de figure ?

**Comment by vfabry** [comment]
Oui, je reproduis : si toutes les couches du theme sont off, alors le compteur affiche 0/1, même s'il y plus d'une couche dans le thème 

![Image](https://github.com/user-attachments/assets/b76f7f2b-e033-4349-9b05-d5cd0221ec98)
